aea/decision_maker/base.py

```diff
@@ -301,11 +306,14 @@ def _handle_tx_message(self, tx_message: TransactionMessage) -> None:
         :param tx_message: the transaction message
         :return: None
         """
-        score_diff = self.preferences.get_score_diff_from_transaction(self.ownership_state, tx_message)
-        if score_diff >= 0.0:
-            pass  # TODO: implement tx flow
-            # envelope = Envelope()
-            # self.outbox.put(envelope)
+        m_address = generate_address_from_public_key(self._wallet.public_keys['fetchai'])
+        api = LedgerApi("127.0.0.1", 8000)
+        if tx_message.get("amount") <= api.tokens.balance(m_address):
+            m_entity = Entity.from_hex(self._wallet.crypto_objects['fetchai'].private_key)
```

Comment:
we should make the entity a property of the wallet I think, like public keys, addresses etc.

Reply:
No, no. Because through the entity you can access the private key.